network/core-node/admin-guide/monitoring.mdx

```diff
@@ -172,15 +172,46 @@ The output will look something like:
+By default, a node will relay or respond to a survey message if the message originated from a node in the receiving node's transitive quorum. This behavior can be overridden by setting the `SURVEYOR_KEYS` field in the config file to a more restrictive set of nodes to relay or respond to. Set `SURVIVOR_KEYS` to `["$self"]` to opt-out of responding to survey requests entirely.
 
 The survey works in two phases: the collecting phase, and the reporting phase. During the collecting phase, nodes record information about themselves and their peers, such as the number of messages sent to a given peer. During the reporting phase, the surveyor requests the results of the collecting phase from nodes on the network.
 
-The surveyor begins the collecting phase by broadcasting a `TimeSlicedSurveyStartCollectingMessage`. The surveyor ends the collecting phase and initiates the reporting phase by broadcasting a `TimeSlicedSurveyStopCollectingMessage`. These start/stop collecting messages ensure that the collecting phase is roughly equal for all nodes present for the duration of the collecting phase.
+The surveyor begins the collecting phase by broadcasting a `TimeSlicedSurveyStartCollectingMessage`. The surveyor ends the collecting phase and initiates the reporting phase by broadcasting a `TimeSlicedSurveyStopCollectingMessage`. These "start/stop collecting" messages ensure that the collecting phase is roughly equal in duration for all nodes present during the entire collecting phase. We recommend sending the "stop collecting" message about 20 minutes after the "start collecting" message. If 30 minutes elapse without receiving a "stop collecting" message, the survey will automatically transition to the reporting phase.
+
+Additionally, the "stop/start collecting" messages contain a `nonce` field identifying the survey instance. The nonce in the "stop collecting" message must match the nonce from the "start collecting" message. The surveyor should choose a random 32-bit unsigned integer for the nonce.
 
 During the reporting phase, the surveyor sends `TimeSlicedSurveyRequestMessage`s to individual nodes to gather the information the node recorded during the collecting phase.
 
-#### Example Survey Command
+#### Overlay Survey Script
+
+To simplify running an overlay survey, stellar-core ships with a script `OverlaySurvey.py` in the `scripts` directory. This script walks the network using the overlay survey HTTP endpoints to build a graph containing the topology of the overlay network. The script outputs this graph both in JSON format, as well as GraphML.  You can analyze the GraphML file using a GraphML viewer such as [Gephi](https://gephi.org/).
+
+An example usage of the survey script to run an overlay survey is as follows:
+```bash
```

Comment:
I'm running into issues with running this script. 

```
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=11626): Max retries exceeded with url: /startsurveycollecting?nonce=483752766 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x100ec0aa0>: Failed to establish a new connection: [Errno 61] Connection refused'))
```

and errors regarding modules not found (in the docker terminal)
Are you able to add me on discord to continue debugging: thegodhand_

Reply:
Just sent you a discord message